qemu/tests/cpu_device_hotpluggable.py

```diff
@@ -117,11 +117,25 @@ def sub_pause_resume():
     check_cpu_topology = params.get_boolean("check_cpu_topology", True)
 
     vm = env.get_vm(params["main_vm"])
+    maxcpus = vm.cpuinfo.maxcpus
+    if not params.objects("vcpu_devices"):
+        vcpus_count = (vm.cpuinfo.threads if
+                       params["machine_type"].startswith("pseries") else 1)
+        pluggable_cpus = vm.cpuinfo.maxcpus // vcpus_count // 2
+        params["vcpu_devices"] = " ".join(["vcpu%d" % (count + 1) for count in
+                                           range(pluggable_cpus)])
+        vm.destroy()
+        if len(params.objects("vcpu_devices")) < 2:
+            test.cancel("Insufficient maxcpus for multi-CPU hotplug")
+        params["paused_after_start_vm"] = "no"
+        vm.create(params=params)
+
+    if vm.is_paused():
```

Comment:
Would you pls tell in which scenario vm is paused? I thought vm is always running at this point.

Reply:
This is to be compatible vcpu_devices is not empty.
For instance, `vcpu_devices = vcpu1 vcpu2`, then the previous code will be skipped, so the VM is launched and in the pause status, we need to resume it.